### ISSUE TITLE
Possible fix for methods that are failing because of SageMaker Python SDK 1.x deprecation

### ISSUE THREAD

**Issue by arshpunia** [aws-samples/aws-ml-detection-workshop#4]
Hello, 

_Tl;dr In light of code changes introduced in SageMaker Python SDK v2.x, some cells of the Jupyter notebook are returning warnings/errors. I have adapted the given notebook to SDK v2.x. If you are accepting PR's, I'd be happy to send one your way._ 

Some methods in the given Jupyter notebook throw warnings (exhibit A) or fail (exhibit B) because of code changes introduced in SageMaker Python SDK v2.x

Exhibit A: 

Upon running the following code snippet, I get a deprecation warning (Exact warning after the code snippet)
```
# Set up the estimator with training job configuration
ip_insights = sagemaker.estimator.Estimator(
    image, 
    execution_role, 
    train_instance_count=1,
    ##instance_count=1,
    train_instance_type='ml.m4.xlarge',
    ##instance_type='ml.p3.2xlarge',
    output_path='s3://{}/{}/output'.format(bucket, prefix),
    sagemaker_session=sagemaker.Session())

# Configure algorithm-specific hyperparameters
ip_insights.set_hyperparameters(
    num_entity_vectors='20000',
    random_negative_sampling_rate='5',
    vector_dim='128', 
    mini_batch_size='10000',
    epochs='5',
    learning_rate='0.01',
)

# Start the training job (should take 3-4 minutes to complete)  
ip_insights.fit(input_data)
```

Here is the warning: 
```
train_instance_count has been renamed in sagemaker>=2.
See: https://sagemaker.readthedocs.io/en/stable/v2.html for details.
train_instance_type has been renamed in sagemaker>=2.
See: https://sagemaker.readthedocs.io/en/stable/v2.html for details.
```

Exhibit B

When running the following code snippet, the cell fails (exact error message given after the snippet)

```
predictor = ip_insights.deploy(
    initial_instance_count=1,
    instance_type='ml.m4.xlarge'
)

from sagemaker.predictor import csv_serializer, json_deserializer

predictor.content_type = 'text/csv'
predictor.serializer = csv_serializer
predictor.accept = 'application/json'
predictor.deserializer = json_deserializer
```

The error message is:

```
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-17-66243338142c> in <module>
      1 from sagemaker.predictor import csv_serializer, json_deserializer
      2 
----> 3 predictor.content_type = 'text/csv'
      4 predictor.serializer = csv_serializer
      5 predictor.accept = 'application/json'

AttributeError: can't set attribute

```

My investigation revealed that content types are now set during method deployment. [Source](https://sagemaker.readthedocs.io/en/stable/v2.html#serializer-and-deserializer-in-create-model)

I have a modified notebook that addresses the code changes made in SDK v2.x. and runs with no known issues.
Are you folks accepting pull requests? If yes, I'd be happy to send the notebook your way.

Please let me know if I can provide any more details. Cheers!

**Comment by ewaldvzyl** [comment]
@arshpunia would you mind sharing these code fixes with me?

I'm facing the same issue with the content type being deprecated and I've tried implementing the new syntax but it raises more and more errors.

Any assistance would be greatly appreciated
